programs/launcher/main.cpp

```diff
@@ -458,8 +458,8 @@ launcher_def::initialize (const variables_map &vmap) {
   if (prod_nodes > total_nodes)
     total_nodes = prod_nodes;
 
-  char * erd = getenv ("EOS_ROOT_DIR");
-  if (erd == 0) {
+  erd = getenv ("EOS_ROOT_DIR");
```

Comment:
`getenv()` can return NULL. And storing NULL in `std::string` is undefined: 

> basic_string( const CharT* s, const Allocator& alloc = Allocator() );
The behavior is undefined if s does not point at an array of at least Traits::length(s)+1 elements of CharT, including the case when s is a null pointer.